src/components/Menus/DropdownMenu.js

```diff
@@ -0,0 +1,207 @@
+    children: PropTypes.node.isRequired,
+    /** React component or HTML that will be used as the dropdown trigger. ie. Button or Icon */
+    triggerElement: PropTypes.element,
+    /** Open and close menu manually */
+    open: PropTypes.bool,
+    /** Callback fired on dropdown close */
+    onClose: PropTypes.func,
+    /** Callback fired on click */
+    onClick: PropTypes.func,
+    /** Callback fired on dropdown open */
+    onOpen: PropTypes.func,
+    /** Callback function fired on key down */
+    onKeyDown: PropTypes.func,
+    /** Requesting to open/close for controlled open component. */
+    onRequestChange: PropTypes.func,
+    /** Callback function fired when a MenuItem is selected */
+    onSelect: PropTypes.func,
+    /** Customize style of menu parent */
+    style: PropTypes.shape({}),
+    /** Props passed to Menu component */
+    menuProps: PropTypes.shape({})
+  }
+
+  static defaultProps = {
+    open: null
+  }
+
+  state = {
+    open: false
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open !== null && this.props.open !== nextProps.open) {
+      if (nextProps.open) {
+        this.open()
+      } else {
+        this.close()
+      }
+    }
+  }
+
+  _handleSelect = (option) => {
+    const { onSelect, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(false)
+    } else {
+      // if multi then do not close
+      this.close()
+    }
+
+    onSelect && onSelect(option)
+  }
+
+  _handleClick = (event) => {
+    event.preventDefault()
+    const { open, onClick, onRequestChange } = this.props
+
+    if (this.controlledOpen()) {
+      onRequestChange && onRequestChange(!open)
+      return
+    }
+
+    if (!this.state.open) {
+      this.open()
+    } else {
+      this.close()
+    }
+    onClick && onClick(event)
+  }
+
+  _handleMenuBlur = () => {
+    const { onRequestChange } = this.props
+    // Timeout ensures click event has correct state when attempting to close
+    setTimeout( () => {
+      if (this.controlledOpen()) {
+        onRequestChange && onRequestChange(false)
+      } else {
+        this.close()
+      }
+    }, 100)
+  }
+
+  _handleKeyDown = (event) => {
+    const { onRequestChange } = this.props
+    switch(event.key) {
+      case 'Enter':
+        if (!event.defaultPrevented && this.controlledOpen()) {
+          event.preventDefault()
+          onRequestChange && onRequestChange(true)
+        } else {
+          event.preventDefault()
+          if (this.state.open) {
+            this.close()
+          } else {
+            this.open()
+          }
+        }
+        break
+      case 'Escape':
+        event.preventDefault()
+
+        if (this.controlledOpen()) {
+          onRequestChange && onRequestChange(false)
+        } else {
+          this.state.open && this.close()
+        }
+        break
+    }
+  }
+
+  controlledOpen() {
+    return this.props.open !== null
+  }
+
+  open() {
```

Comment:
Maybe we should stick with the handler convention of prefixing this with `handle`?

Reply:
I would favor prefixing with `on` instead of `handle`. Shorter and sweeter.